### PR TITLE
[16.0] stock_release_channel: Manage channel lifecycle

### PR DESCRIPTION
**Pull request by lmignon**
* [x] includes #493
* [x] depends on #478 

The availability of a release channel depends on its state. A release channel
can be in one of the following states:

  - Open: the channel is available and can be used to release transfers. New
    transfer are assigned automatically to this channel.
  - Locked: the channel is available but the release of transfers from the channel
    is no more possible. New transfers are still automatically assigned to this
  - Asleep: the channel is not available and cannot be used to release
    transfers. It is also no more possible to assign transfers to this channel.

New release channels are by default "Open". You can change its state by using
the "Lock" and "Sleep" buttons. When the "Sleep" button is used, in addition to
the state change, not completed transfers assigned to the channel are unassigned
from the channel. When the "Lock" button is used, only the state change is changed.
A locked channel can be unlocked by using the "Unlock" button.
A asleep channel can be waken up by using the "Wake up" button. When the "Wake up"
button is used, in addition to the state change, the system looks for pending
transfers requiring a release and try to assign them to a channel in the
"Open" or "Locked" state.

### CODE REVIEW

File: stock_release_channel/models/stock_release_channel.py [hunk 1]
```diff
@@ -621,3 +705,72 @@ def release_next_batch(self):
...
+    def action_unlock(self):
+        self._check_is_action_unlock_allowed()
+        self.write({"state": "open"})
+
+    def action_sleep(self):
+        self._check_is_action_sleep_allowed()
+        pickings_to_unassign = self.env["stock.picking"].search(
+            self._get_picking_to_unassign_domain()
+        )
+        pickings_to_unassign.write({"release_channel_id": False})
```
Comment (jbaudoux) [inline stock_release_channel/models/stock_release_channel.py:788]: Before unassigning, the picking must be "unreleased".
Could be some ongoing pickings cannot be unreleased and this action will be declined

Comment (lmignon) [reply stock_release_channel/models/stock_release_channel.py:784]: This PR doesn't depend on #478 I'll implement this requirement in #508

Comment (lmignon) [reply stock_release_channel/models/stock_release_channel.py:788]: @jbaudoux done

**Comment by lmignon** [comment]
@simahawk @jbaudoux rebase done but  requires #478

**Review by rousseldenis — APPROVED** [review]
Code review.

@lmignon Could you remove the test-requirements commit ?

**Comment by lmignon** [comment]
@rousseldenis test-requirements removed

**Review by jbaudoux — APPROVED** [review]
LG

**Pull request merged**